// filter-material-ui/lib/FilterMaterialUi.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TYPE = void 0;
var react_1 = __importDefault(require("react"));
var TextFilter_1 = __importDefault(require("./TextFilter"));
var FormFilter_1 = __importDefault(require("./FormFilter"));
var react_2 = require("react");
function FilterMaterialUi(props) {
    var onChange = props.onChange, textPrefix = props.textPrefix, textSuffix = props.textSuffix, rest = __rest(props, ["onChange", "textPrefix", "textSuffix"]);
    var _a = (0, react_2.useState)(true), isTextVisible = _a[0], setTextVisible = _a[1];
    var handleEdit = (0, react_2.useCallback)(function () { return setTextVisible(false); }, []);
    var handleShowText = (0, react_2.useCallback)(function () { return setTextVisible(true); }, []);
    var handleChange = (0, react_2.useCallback)(function (data) {
        handleShowText();
        onChange(data);
    }, []);
    return isTextVisible ? (react_1.default.createElement(TextFilter_1.default, __assign({}, rest, { textPrefix: textPrefix, textSuffix: textSuffix, onEdit: handleEdit }))) : (react_1.default.createElement(FormFilter_1.default, __assign({}, rest, { onCancel: handleShowText, onChange: handleChange })));
}
var TYPE;
(function (TYPE) {
    TYPE[TYPE["INPUT"] = 0] = "INPUT";
    TYPE[TYPE["SINGLE_SELECT"] = 1] = "SINGLE_SELECT";
    TYPE[TYPE["MULTIPLE_SELECT"] = 2] = "MULTIPLE_SELECT";
    TYPE[TYPE["COLORS_SELECT"] = 3] = "COLORS_SELECT";
})(TYPE = exports.TYPE || (exports.TYPE = {}));
exports.default = FilterMaterialUi;
